tests/lit-tests/discard-float16.ispc

```diff
@@ -0,0 +1,7 @@
+// RUN: not %{ispc} --target=host --nostdlib --nowrap -o - %s 2>&1 | FileCheck %s
+
+// CHECK: Error: syntax error
+// CHECK-NOT: detected memory leaks
```

Comment:
It's just compiled with ISPC, but not executed with ASAN, so where "detected memory leaks" would come from?

Reply:
Yeah, Nowhere. This test is specifically target asan build with `CHECK-NOT: detected memory leak`. Otherwise, it is quite useless.

I actually need to distinguish two states:
1. `ispc` exited with non-zero because of syntax errors,
2. `ispc` exited with non-zero because of memory leak.

The first one need to be `PASS`, the second one needs to be `FAIL` for tests in this PR. Maybe, the better approach is required.

Reply:
You can try to detect `ASAN_OPTIONS` environment variable in `lit.cfg`, add it to available lit features `config.available_features.add("ASAN_BUILD")` and use it from the lit tests to enable them only for ASAN build.

Reply:
I don't think that the tests are useless without ASAN, they specifically checking that syntax error is generated. So possibly it's fine running without detecting ASAN, I'm good with either way of running them.

Reply:
> You can try to detect `ASAN_OPTIONS` environment variable in `lit.cfg`, add it to available lit features `config.available_features.add("ASAN_BUILD")` and use it from the lit tests to enable them only for ASAN build.

I have added `ASAN_RUN` and `REQUIRES: ASAN_RUN` to tests.  But these tests are `UNSUPPORTED` without `ASAN_RUN`. Is there some way to run them always and does `CHECK-NOT` only when `ASAN_RUN`?

Reply:
I think the only way is to split test into two. Always check that syntax error is generated in one test, and check for memory leaks in another.

Reply:
OK, I have duplicated them.